packages/eslint-plugin/test/test-no-nested-await.js

```diff
@@ -0,0 +1,31 @@
+/* eslint-env node */
+
+'use strict';
+
+// Convenience utility using Mocha globals.
+const { RuleTester } = require('eslint');
+
+const ruleTester = new RuleTester({
+  parserOptions: { sourceType: 'module', ecmaVersion: 2021 },
+});
+const rule = require('../lib/rules/no-nested-await.js');
+const {
+  clearlyBalanced,
+  subtlyBalanced,
+  clearlyUnbalanced,
+} = require('./corpus.js');
+
+const unexpectedNestedAwait = 'unexpectedNestedAwait';
+
+ruleTester.run('no-nested-await', rule, {
+  valid: clearlyBalanced,
+  // `no-nested-await` is not smart enough to ignore subtle code that is
+  // actually balanced.  We test that it fails these cases.
+  invalid: [...clearlyUnbalanced, ...subtlyBalanced].map(example => ({
```

Comment:
I appreciate the comment. Especially on `subtlyBalanced` in the other file. What the comment here says may be obvious if the names were mapped to this "smart enough" aspect:

```suggestion
  valid: trueValid,
  invalid: [...trueInvalid, ...falseInvalid].map(example => ({
```

Reply:
I like `clearly*` vs `subtly*`, but I agree that `Balanced` vs. `Unbalanced` should be `Valid` vs. `Invalid`.